nixos/modules/services/continuous-integration/gitlab-runner.nix

```
@@ -499,29 +499,29 @@ in {
             '';
           };
           preGetSourcesScript = mkOption {
-            type = types.nullOr types.path;
+            type = types.nullOr (types.either types.str types.path);
```

Isn't a valid path always a valid string? So we could just do

```suggestion
            type = types.nullOr types.path;
```